plugins/pulumi/pulumi_access_token.go

```diff
@@ -0,0 +1,93 @@
+		DocsURL:       sdk.URL("https://www.pulumi.com/docs/intro/pulumi-service/accounts/"),
+		ManagementURL: sdk.URL("https://app.pulumi.com/account/tokens"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to Pulumi.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 44,
+					Prefix: "pul-",
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Host,
+				MarkdownDescription: "The Pulumi host to authenticate to. Defaults to 'app.pulumi.com'.",
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			importer.TryEnvVarPair(map[string]sdk.FieldName{
+				"PULUMI_BACKEND_URL":  fieldname.Host,
+				"PULUMI_ACCESS_TOKEN": fieldname.Token,
+			}),
+			TryPulumiConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"PULUMI_ACCESS_TOKEN": fieldname.Token,
+}
+
+// Duplicated from:
+// https://github.com/pulumi/pulumi/blob/874a8de2dac2fe2c8138cf63f3f242660bffc738/sdk/go/common/workspace/creds.go#L112-L127
+// --- START
+// Account holds the information associated with a Pulumi account.
+type Account struct {
+	AccessToken     string    `json:"accessToken,omitempty"`     // The access token for this account.
+	Username        string    `json:"username,omitempty"`        // The username for this account.
+	Organizations   []string  `json:"organizations,omitempty"`   // The organizations for this account.
+	LastValidatedAt time.Time `json:"lastValidatedAt,omitempty"` // The last time this token was validated.
+	Insecure        bool      `json:"insecure,omitempty"`        // Allow insecure server connections when using SSL.
+}
+
+// Credentials hold the information necessary for authenticating Pulumi Cloud API requests.  It contains
+// a map from the cloud API URL to the associated access token.
+type Credentials struct {
+	Current      string             `json:"current,omitempty"`      // the currently selected key.
+	AccessTokens map[string]string  `json:"accessTokens,omitempty"` // a map of arbitrary key strings to tokens.
+	Accounts     map[string]Account `json:"accounts,omitempty"`     // a map of arbitrary keys to account info.
+}
+
+// --- END
+
+// TODO: Check if the platform stores the Access Token in a local config file, and if so,
+// implement the function below to add support for importing it.
+func TryPulumiConfigFile() sdk.Importer {
+	return importer.TryFile("~/.pulumi/credentials.json", func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		var config Credentials
+		if err := contents.ToJSON(&config); err != nil {
+			out.AddError(err)
+			return
+		}
+
+		// out.AddCandidate(sdk.ImportCandidate{
```

Comment:
The importer needs to have an `out.AddCandidate` so that it passes the imported access tokens to 1Password.

What "arbitrary keys" are used in the map of access token and accounts? Would those be nice to include in the name of the 1Password item to differentiate between access tokens in case there are multiple configured in this config file? If so, we can use the `NameHint` to have 1Password CLI suggest the key as part of the name of the 1Password item during the import.

Reply:
Implemented with tests!